src/gt4py/next/iterator/ir_utils/ir_makers.py

```diff
@@ -58,7 +60,11 @@ def ref(
     if isinstance(ref_or_name, itir.SymRef):
         assert not type_
         return ref_or_name
-    return itir.SymRef(id=ref_or_name, type=ensure_type(type_))
+    ref = itir.SymRef(id=ref_or_name, type=ensure_type(type_))
+    if annex is not None:
+        for key, value in annex.items():
+            setattr(ref.annex, key, value)
+    return ref
```

Comment:
What is the intended behavior if we pass a `SymRef` together with a non-empty annex? Shouldn't a new `SymRef` be created with the new annex?